docs/front-end/web-components.md

```diff
@@ -0,0 +1,82 @@
+# Web Components
+
+The [WHATWG](https://html.spec.whatwg.org/multipage/custom-elements.html) and [WICG](https://github.com/WICG/webcomponents) maintain suite of web technologies and specifications including (but not limited to)
+
+- [Custom Elements](https://html.spec.whatwg.org/multipage/custom-elements.html#custom-elements)
+- [Shadow DOM](https://dom.spec.whatwg.org/#shadow-trees)
+- The [`<template>` element](https://html.spec.whatwg.org/multipage/scripting.html#the-template-element)
+- The [`<slot>` element](https://html.spec.whatwg.org/multipage/scripting.html#the-slot-element)
+- [`ElementInternals`](https://html.spec.whatwg.org/multipage/custom-elements.html#element-internals)
+- [CSS Custom Properties](https://drafts.csswg.org/css-variables/)
+- [Module scripts](https://html.spec.whatwg.org/#module-script)
+
+These are collectively referred to as "web components", and represent the browsers' native component model for web developers.
+
+Broadly speaking, every web component is a custom element - a specific, globally-recognized tag-name which becomes associated with a class extending HTMLElement in JavaScript. Web components can also make use of the related technologies.
```

Comment:
```suggestion
Broadly speaking, every web component is a custom element - a specific tag-name which becomes associated with a class extending HTMLElement in JavaScript. Web components can also make use of the related technologies.
```

[Scoped Custom Element Registries](https://github.com/WICG/webcomponents/blob/gh-pages/proposals/Scoped-Custom-Element-Registries.md) is almost ready for prime time with [webkit support](https://bugs.webkit.org/show_bug.cgi?id=267454) completed and [gecko](https://bugzilla.mozilla.org/show_bug.cgi?id=1874414) not far behind.